lib/jit_preloader.rb

```diff
@@ -25,6 +25,16 @@ def self.globally_enabled=(value)
     @enabled = value
   end
 
+  def self.query_dive_limit=(limit)
```

Comment:
I'm going to make a deeply annoying suggestion, and for that I apologize in advance. 🙂 

Is there a better name for this concept than "dive limit"? That's a very MySQL-specific term which would be confusing to anyone who's a Postgres user or who doesn't know the internals of MySQL as thoroughly as Richard does. Maybe we could make this more readable for other developers by calling it `max_ids_per_query`, or something similar which expresses the effect.

Reply:
@burnison brought this up, too, since `dive limit` may not even be a generic term across different DBs. I can't think of a concise term that can easily express why we constrain the limit of IDs per query. I also considered a similar term like `max_ids_per_query ` before, but I am not sure whether it's apparent to user _why_ we even need to provide that ability instead of just a single query with all IDs.

So my plan is after this, we should update the README to explain the reason and usage of it. But I can get persuaded?

Reply:
Why not both? We could call it `max_ids_per_query` (or any other similar database-independent name which expresses what it does), and include some details in the README to explain why it's there and when to use it. Seems like the best of both worlds?

Reply:
Good question! Depends on the name. `dive_limit` doesn't give the user any guidance on whether they should pass N or N-1. `max_ids_per_query`, on the other hand, is pretty clear that it's the maximum number of IDs we'll put in a query, so it implies N-1 already.

Reply:
Yeah, that's clearer. We can push the responsibility of figuring out the `N-1` number to the user. Sounds good! I will make the changes.